src/libs/events.js

```diff
@@ -106,17 +107,30 @@ const eventsList = {
   workspaceStar: 'workspace:star'
 }
 
+/**
+ * Extracts name, namespace, and cloudPlatform (if present) from an object. The object can either have these
+ * as top-level properties (such as would be returned from parseNav), or nested within a workspace object
+ * (such as would be returned from the ajax workspace details API).
```

Comment:
Note that I decided to support passing the workspace object because a lot of Analyses code passes `workspace` instead of `workspace.workspace` (which means they currently event `undefined` in MixPanel). This is certainly understandable due to the lack of documentation and types, so I decided to make the method forgiving vs. trying to fix all the calling locations.